rad/src/script.rs

```diff
@@ -20,30 +21,115 @@ use crate::{
+    }
+}
+
+impl RadonScriptExecutionSettings {
+    /// Enable all execution features except `partial_results`. This is the default for
+    /// `witnet_node`.
+    pub fn all_but_partial_results() -> Self {
+        Self {
+            partial_results: false,
+            ..Self::enable_all()
+        }
+    }
+
+    /// Disable all execution features.
+    pub fn disable_all() -> Self {
+        Self {
+            partial_results: false,
+            timing: false,
+            breakpoints: false,
+        }
+    }
+
+    /// Enable all execution features. This is recommended for `witnet_wallet`.
+    pub fn enable_all() -> Self {
+        Self {
+            partial_results: true,
+            timing: true,
+            breakpoints: true,
+        }
+    }
+}
+
 /// Run any RADON script on given input data, and return `RadonReport`.
+/// By enabling or disabling each of the specific flags in the settings argument, we can adjust how
+/// much execution metadata we want to track, e.g. execution time, partial results, etc.
 pub fn execute_radon_script(
     input: RadonTypes,
     script: &[RadonCall],
     context: &mut ReportContext,
+    settings: RadonScriptExecutionSettings,
 ) -> Result<RadonReport<RadonTypes>, RadError> {
-    // Set the execution timestamp
-    context.start();
-    // Run the execution
+    // Set the execution start timestamp, if enabled by `timing` setting
+    if settings.timing {
+        context.start();
+    }
+
+    // Initialize a vector for storing the partial results, if enabled by `partial_results` setting
+    let mut partial_results = if settings.partial_results {
+        Some(vec![Ok(input.clone())])
+    } else {
+        None
+    };
+
+    // Run the execution by recursively applying calls into the result of the previous call
     let result = script
         .iter()
         .enumerate()
         .try_fold(input, |input, (i, call)| {
-            context.call_index = Some(u8::try_from(i).unwrap());
-            operate_in_context(input, call, context)
+            // Update the index of the last executed call, if enabled in settings
+            if settings.breakpoints {
+                context.call_index = Some(u8::try_from(i).unwrap());
+            }
+
+            // Apply the call
+            let partial_result = operate_in_context(input, call, context);
+
+            // Keep partial result, if enabled by `partial_results` setting
+            if let Some(partial_results) = partial_results.as_mut() {
+                partial_results.push(partial_result.clone());
+            }
+
+            partial_result
         });
-    // Set the completion timestamp
-    context.complete();
+
+    // Set the completion timestamp, if enabled by `timing` settings
+    if settings.timing {
+        context.complete();
+    }
 
     // Return a report as constructed from the result and the context
-    Ok(RadonReport::from_result(result, context))
+    Ok(if settings.partial_results {
```

Comment:
You can also use `let Some` here
```suggestion
    Ok(if let Some(partial_results) = partial_results {
```

Reply:
True, applied!